test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -1169,6 +1209,50 @@ def forward(self, x):
             match_nodes = 12
             self._test_common(mod, (v,), match_count, match_nodes, rtol=1e-2, atol=1e-2)
 
+    def test_qconv2d_maxpool2d_linear_dynamic(self):
+        r"""
+        This testcase will quantize a single Conv2d->Maxpool2d->Linear module
+        with dynamic batch size input.
+        """
+
+        class M(torch.nn.Module):
+            def __init__(
+                self,
+                **kwargs,
+            ):
+                super().__init__()
+                self.conv = torch.nn.Conv2d(
+                    3, 16, (2, 2), stride=(1, 1), padding=(1, 1)
+                )
+                self.relu = torch.nn.ReLU()
+                self.maxpool2d = torch.nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
+                self.avgpool = torch.nn.AdaptiveAvgPool2d((1, 1))
+                self.linear = torch.nn.Linear(16, 16)
+
+            def forward(self, x):
+                temp = self.relu(self.conv(x))
+                temp = self.maxpool2d(temp)
+                temp = self.avgpool(temp)
+                temp = torch.flatten(temp, 1)
+                return self.linear(temp)
+
+        mod = M().eval()
+        v = torch.randn((2, 3, 8, 8), dtype=torch.float32, requires_grad=False).add(1)
```

Comment:
Not sure if I missed something but where is the dynamic shape specified?

Reply:
I think here `dynamic shape` actually means supporting `different batch size inputs` without recompilation. Previously, the generated code will re-compile when inputs have changed batch size since the batch size value has been specialized in generated code. So, in this UT, we will check there are symbolic variable in the generated code instead of all the values are specialized to specific value.